OpenRA.Mods.Common/AI/SupportPowerDecision.cs

```diff
@@ -72,6 +72,14 @@ public int GetAttractiveness(WPos pos, Player firedBy)
 				var checkActors = world.FindActorsInCircle(pos, radiusToUse);
 				foreach (var scrutinized in checkActors)
 					answer += consideration.GetAttractiveness(scrutinized, firedBy.Stances[scrutinized.Owner], firedBy);
+
+				var delta = new WVec(radiusToUse, radiusToUse, WDist.Zero);
+				var tl = world.Map.CellContaining(pos - delta);
+				var br = world.Map.CellContaining(pos + delta);
+				var checkFrozen = frozenLayer.FrozenActorsInRegion(new CellRegion(world.Map.Grid.Type, tl, br));
+				foreach (var scrutinized in checkFrozen)
+					if (scrutinized.IsValid)
```

Comment:
Isn't this missing a `Visible` check, like the below method?

Reply:
No, `Visible` is checked inside `GetAttractiveness`.  The `IsValid` check here is to make sure that `Owner` is not null.  I have added a comment explaining this.